libursa/src/encryption/symm/aescbc.rs

```diff
@@ -0,0 +1,141 @@
+use super::Encryptor;
+use aead::{
+    generic_array::{
+        typenum::{Unsigned, U0, U16, U32, U48, U64},
+        GenericArray,
+    },
+    Aead, Error, NewAead, Payload,
+};
+use aes::{Aes128, Aes256};
+use block_modes::block_padding::Pkcs7;
+use block_modes::{BlockMode, Cbc};
+use hmac::{Hmac, Mac};
+#[cfg(feature = "serialization")]
+use serde::de::{Deserialize, Deserializer, Error as DError, Visitor};
+#[cfg(feature = "serialization")]
+use serde::ser::{Serialize, Serializer};
+use sha2::{Sha256, Sha512};
+use subtle::ConstantTimeEq;
+use zeroize::Zeroize;
+
+type Aes128Cbc = Cbc<Aes128, Pkcs7>;
+type Aes256Cbc = Cbc<Aes256, Pkcs7>;
+type HmacSha256 = Hmac<Sha256>;
+type HmacSha512 = Hmac<Sha512>;
+
+macro_rules! aes_cbc_hmac_impl {
+    ($name:ident, $algokeysize:ident, $keysize:ident, $noncesize:ident, $tagsize:ident, $algo:ident, $hash:ident, $visitor:ident) => {
```

Comment:
What is `$algokeysize`?

Reply:
https://github.com/hyperledger/ursa/pull/76/files#diff-49cadd47a44ba93f412f87ba4cc2427fR57

Reply:
The reason I asked was because I did not understand the logic. Can you point to some link (outside this PR works too) where it is described?

Reply:
This is just a macro. The caller doesn't know the difference. If you want to understand the logic, then look at the code where its used.